zerver/lib/actions.py

```diff
@@ -4830,17 +4831,19 @@ def do_change_default_events_register_stream(
             stream_name: Optional[str] = stream.name
         else:
             stream_name = None
-        send_event(
-            user_profile.realm,
-            dict(
-                type="realm_bot",
-                op="update",
-                bot=dict(
-                    user_id=user_profile.id,
-                    default_events_register_stream=stream_name,
+        transaction.on_commit(
+            lambda: send_event(
+                user_profile.realm,
+                dict(
+                    type="realm_bot",
+                    op="update",
+                    bot=dict(
+                        user_id=user_profile.id,
+                        default_events_register_stream=stream_name,
+                    ),
                 ),
-            ),
-            bot_owner_user_ids(user_profile),
+                bot_owner_user_ids(user_profile),
+            )
         )
```

Comment:
Would it be better to construct the `event` object outside the `transaction.on_commit` block, so that it's just the `send_event` call that sits in there?